common/src/main/java/org/opensearch/ml/common/transport/connector/MLCreateConnectorInput.java

```diff
@@ -248,6 +266,13 @@ public void writeTo(StreamOutput output) throws IOException {
         }
         output.writeBoolean(dryRun);
         output.writeBoolean(updateConnector);
+        if (connectorClientConfig != null) {
```

Comment:
Maybe add a version filter here to avoid bwc issue?

Reply:
We have default values [here](https://github.com/opensearch-project/ml-commons/blob/1dd9f344484be9464b9c937b7ad32ab3ea6b44a6/common/src/main/java/org/opensearch/ml/common/connector/ConnectorClientConfig.java). Could you please explain a bit more why BWC issue could happen here?

Reply:
It's not because if there is value or not, it's because you are introduce new field here. If a master node with this version sends this input to a older data node, it will throw an exception due to cannot process